ui/desktop/electron-app/src/session/manager.js

```diff
@@ -0,0 +1,51 @@
+const Session = require('./index.js');
+
+class SessionManager {
+  #sessions;
+
+  constructor() {
+    this.#sessions = [];
+  }
+
+  /**
+   * Start a session and track it.
+   * Returns session proxy details if successfully started.
+   * Only successfully started sessions will be tracked.
+   * @param {string} addr
+   * @param {string} target_id
+   * @param {string} token
+   * @param {string} host_id
+   */
+  async start(addr, target_id, token, host_id) {
+    const session = new Session(addr, target_id, token, host_id);
+    return session.start().then(() => {
```

Comment:
Suggestion:
```
this.#sessions.push(session);
return session.start(); // the start promise could return proxy details.
```

Reply:
My approach was to only track successful session processes so that I could cancel all of them on quitting the app with a simple loop. I will reconfigure this to check for an active session. It'll simplify code as you suggested.